src/Microsoft.Health.Fhir.CosmosDb/Registration/FhirServerBuilderCosmosDbRegistrationExtensions.cs

```diff
@@ -101,11 +103,15 @@ private static IFhirServerBuilder AddCosmosDbPersistence(this IFhirServerBuilder
                 .AsService<ICollectionInitializer>();
 
             services.Add<FhirCollectionSettingsUpdater>()
-                .Singleton()
+                .Transient()
```

Comment:
Changing these to transient will cause a new instance for each request, correct?  Is that necessary for these types?

Reply:
I might review again, but these are only used once as far as I know. Having them as singleton means the container will hold onto the reference